lib/src/http/auto_retry.dart

```diff
@@ -154,8 +161,27 @@ class RequestAutoRetry extends AutoRetryConfig {
     if (request is MultipartRequest && request.files.isNotEmpty) return false;
     return true;
   }
+
+  /// timeoutThreshold will not move beyond 60s or the [_request.timeoutThreshold], whichever is greater, when [increaseTimeoutOnRetry] is true.
+  Duration get timeoutThreshold {
+    if (increaseTimeoutOnRetry) {
+      return getRetryTimeoutThreshold(_request.timeoutThreshold, numAttempts);
+    }
+
+    return _request.timeoutThreshold;
+  }
 }
 
+@visibleForTesting
+Duration getRetryTimeoutThreshold(Duration timeoutThreshold, int numAttempts) {
+    if (numAttempts <= 0) return timeoutThreshold;
+    var threshold = timeoutThreshold * numAttempts;
+    var maxTimeout = max<int>(timeoutThreshold.inSeconds, 60);
+    threshold = Duration(seconds: min(threshold.inSeconds, maxTimeout));
+  return threshold;
+}
```

Comment:
I did this because I could test this pure function without have to set up a bunch of mocks.